Tests/Functional/SelectedTest/main.swift

```diff
@@ -0,0 +1,58 @@
+// RUN: %{swiftc} %s -o %{built_tests_dir}/SelectedTest
+// RUN: %{built_tests_dir}/SelectedTest SelectedTest.ExecutedTestCase/test_foo > %t || true
+// RUN: %{built_tests_dir}/SelectedTest SelectedTest.ExecutedTestCase >> %t || true
+// RUN: %{built_tests_dir}/SelectedTest >> %t || true
+// RUN: %{xctest_checker} %t %s
+//// Output from the first invocation (running a single test method)
+// CHECK: Test Case 'ExecutedTestCase.test_foo' started.
+// CHECK: Test Case 'ExecutedTestCase.test_foo' passed \(\d+\.\d+ seconds\).
+// CHECK: Executed 1 test, with 0 failures \(0 unexpected\) in \d+\.\d+ \(\d+\.\d+\) seconds
+// CHECK: Total executed 1 test, with 0 failures \(0 unexpected\) in \d+\.\d+ \(\d+\.\d+\) seconds
+//// Output from the second invocation (running one entire test case)
+// CHECK: Test Case 'ExecutedTestCase.test_bar' started.
+// CHECK: Test Case 'ExecutedTestCase.test_bar' passed \(\d+\.\d+ seconds\).
+// CHECK: Test Case 'ExecutedTestCase.test_foo' started.
+// CHECK: Test Case 'ExecutedTestCase.test_foo' passed \(\d+\.\d+ seconds\).
+// CHECK: Executed 2 tests, with 0 failures \(0 unexpected\) in \d+\.\d+ \(\d+\.\d+\) seconds
+// CHECK: Total executed 2 tests, with 0 failures \(0 unexpected\) in \d+\.\d+ \(\d+\.\d+\) seconds
+//// Output from the third invocation (running all tests)
+// CHECK: Test Case 'ExecutedTestCase.test_bar' started.
+// CHECK: Test Case 'ExecutedTestCase.test_bar' passed \(\d+\.\d+ seconds\).
+// CHECK: Test Case 'ExecutedTestCase.test_foo' started.
+// CHECK: Test Case 'ExecutedTestCase.test_foo' passed \(\d+\.\d+ seconds\).
+// CHECK: Executed 2 tests, with 0 failures \(0 unexpected\) in \d+\.\d+ \(\d+\.\d+\) seconds
+// CHECK: Test Case 'SkippedTestCase.test_baz' started.
+// CHECK: Test Case 'SkippedTestCase.test_baz' passed \(\d+\.\d+ seconds\).
+// CHECK: Executed 1 test, with 0 failures \(0 unexpected\) in \d+\.\d+ \(\d+\.\d+\) seconds
+// CHECK: Total executed 3 tests, with 0 failures \(0 unexpected\) in \d+\.\d+ \(\d+\.\d+\) seconds
```

Comment:
I've been thoroughly convinced by your suggestions in #43, and I think you should place these assertions as close to the source code they're testing as possible (like this: https://github.com/modocache/swift-corelibs-xctest/commit/22f1d24a0a48dd278d133f18a0252ed91967b289#diff-38abae29a59be553f8b7fc357a30e6ecR14). In fact, I think we should modify all of our tests in this style!